pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1576,15 +1576,9 @@ async def run(self):
                     content_q_by_type[model_type] = content_q_by_type[model_type] | unit_q
                     d_content_by_nat_key[d_content.content.natural_key()].append(d_content)
 
+            saved_rslts_pids = []  # track pulp_ids of anything we specifically save below
+            saved_rslts_by_pid = {}  # hold the to-be-saved results by-pid
```

Comment:
I'm not a fan of these names, because the content have been modified but have not been (re)saved.  Maybe `modified_contents_by_pk`?  Also if you're using the PK as the ID, you don't need to store them in a separate list, you can just use `.keys()`.

Reply:
Can do.